src/Common/Factory/FactoryCreateContainOneNewRule.php

```diff
@@ -110,15 +114,43 @@ protected function isIndirectFactoryMethod(MethodNode $method)
             return false;
         }
 
-        if ($primaryPrefixes[0]->getChild(0)->getName() === '$this' && substr($primaryPrefixes[0]->getChild(1)->getName(), 0, 6) === 'create') {
+        $firstPrimaryPrefix = $primaryPrefixes[0];
+
+        if ($firstPrimaryPrefix->getChild(0)->getName() === '$this' && substr($firstPrimaryPrefix->getChild(1)->getName(), 0, 6) === 'create') {
             return true;
         }
-        if ($primaryPrefixes[0]->getChild(0)->getName() === '$this' && substr($primaryPrefixes[0]->getChild(1)->getName(), 0, 2) === '->') {
+
+        if ($firstPrimaryPrefix->getChild(0)->getName() === '$this' && substr($firstPrimaryPrefix->getChild(1)->getName(), 0, 2) === '->') {
             if (substr($primaryPrefixes[1]->getChild(0)->getName(), 0, 6) === 'create') {
                 return true;
             }
         }
 
+        if ($firstPrimaryPrefix->getParent()->getName() === 'return' &&
+            $firstPrimaryPrefix->getChild(1)->getNode()->getImage() === 'create' &&
+            $this->isStaticMethodCall($firstPrimaryPrefix)
+        ) {
+            return true;
+        }
+
         return false;
     }
+
+    private function isStaticMethodCall(AbstractNode $methodCall)
```

Comment:
The code here was get from https://github.com/phpmd/phpmd/blob/f1c145e538d7cf8c2d1a45fd8fb723eca64005f4/src/main/php/PHPMD/Rule/CleanCode/StaticAccess.php